manager/assets/modext/widgets/core/modx.panel.js

```diff
@@ -415,6 +415,177 @@ Ext.extend(MODx.FormPanel,Ext.FormPanel,{
             }
         });
     }
+
+    /**
+     * @property {Function} onChangeStaticSource - Updates the static file field based
+     * on the chosen source.
+     *
+     * @param {Object} cmp - The media source field's Ext.Component object
+     * @param {String} elType - The MODX element type (i.e., tv, chunk, or snippet)
+     */
+    ,onChangeStaticSource: function(cmp, elType) {
+
+        const   staticFileField = Ext.getCmp(`modx-${elType}-static-file`),
+                staticFile = staticFileField.getValue(),
+                staticDir = staticFile.slice(0, (staticFile.lastIndexOf('/') + 1)),
+                staticFileFieldId = staticFileField.id,
+                staticFileFieldContainer = Ext.getCmp(staticFileField.ownerCt.id),
+                itemKey = staticFileFieldContainer.items.keys.indexOf(staticFileFieldId),
+                previousSource = this.previousFileSource || 0,
+                currentSource = cmp.getValue(),
+                currentRecord = {
+                    static_file: staticFile,
+                    source: currentSource,
+                    openTo: staticDir
+                }
+        ;
+        let newStaticFile,
+            changeFieldType = false,
+            updateFieldSource = false
+        ;
+        if (elType === 'template') {
+            // need these in method's global scope, so define with var instead of const/let
+            var   staticPreviewFileField = Ext.getCmp(`modx-${elType}-preview-file`),
```

Comment:
Either stick with var or const/let

Reply:
This is one place where it has to be var because of how block scoping works with const/let.

Reply:
ugh, ok, this is not ideal approach, but if I consider the rest of the codebase... meh